File: src/main/java/model/updateScores/FourThreeTwoOnePoints.java
package model.updateScores;

import model.CellState;
import model.LeapFrog;

public class FourThreeTwoOnePoints implements UpdateScores {
    private LeapFrog game;

    public FourThreeTwoOnePoints(LeapFrog c) {
        this.game = c;
    }

    public void updateScore(CellState piece) {
        int points = piece.ordinal() + 1;
        int old = this.game.getScores().get(this.game.getTurn());
        this.game.getScores().put(this.game.getTurn(), old + points);
    }
}

File: docs/scripts/updateScores/FourThreeTwoOnePoints.js
import CellState from "../CellState.js";

export default class FourThreeTwoOnePoints {
    constructor(c) {
        this.game = c;
    }
    updateScore(piece) {
        let points = Object.keys(CellState).indexOf(piece) + 1;
        this.game.scores[this.game.getTurn()] += points;
    }
}